diff --git a/2.0/include/pgenlib_read.cc b/2.0/include/pgenlib_read.cc
index b139b746..42b171e4 100644
--- a/2.0/include/pgenlib_read.cc
+++ b/2.0/include/pgenlib_read.cc
@@ -24,6 +24,9 @@
 // Uncomment this during e.g. pgenlibr development to enable error-throwing.
 // #include <stdexcept>
 
+#define fseeko _fseeki64
+#define ftello64 _ftelli64
+
 #ifdef __cplusplus
 namespace plink2 {
 #endif
diff --git a/2.0/include/pgenlib_write.cc b/2.0/include/pgenlib_write.cc
index db96d566..c0dc7ef7 100644
--- a/2.0/include/pgenlib_write.cc
+++ b/2.0/include/pgenlib_write.cc
@@ -19,10 +19,17 @@
 #include <assert.h>
 #include <limits.h>
 #include <stdlib.h>
+#ifdef WIN32
+#include <windows.h>
+#else
 #include <unistd.h>  // unlink()
+#endif
 
 #include "plink2_bits.h"
 
+#define fseeko _fseeki64
+#define ftello64 _ftelli64
+
 #ifdef __cplusplus
 namespace plink2 {
 #endif
diff --git a/2.0/include/plink2_bitmap.cc b/2.0/include/plink2_bitmap.cc
index 1b75b439..8e2f5a67 100644
--- a/2.0/include/plink2_bitmap.cc
+++ b/2.0/include/plink2_bitmap.cc
@@ -21,6 +21,9 @@
 #include <limits.h>
 #include <string.h>
 
+#define fseeko _fseeki64
+#define ftello64 _ftelli64
+
 #ifdef __cplusplus
 namespace plink2 {
 #endif
diff --git a/2.0/include/plink2_memory.cc b/2.0/include/plink2_memory.cc
index 490fdc31..1731b272 100644
--- a/2.0/include/plink2_memory.cc
+++ b/2.0/include/plink2_memory.cc
@@ -16,7 +16,12 @@
 
 #include "plink2_memory.h"
 
-#include <unistd.h>  // sysconf()
+#ifdef WIN32
+#include <windows.h>
+#else
+#include <unistd.h>  // unlink()
+#endif
+
 #ifdef __APPLE__
 #  include <sys/sysctl.h>  // sysctl()
 #endif
